src/http/Response.cpp

```diff
@@ -0,0 +1,208 @@
+		file.close();
+	}
+	return ret;
+}
+
+void
+Response::set_response_type(std::string path)
+{
+	std::string type = "";
+	type = path.substr(path.rfind(".") + 1, path.size() - path.rfind("."));
+	if (type == "html")
+		_response_map["Content-Type"] = "text/html";
+	else if (type == "css")
+		_response_map["Content-Type"] = "text/css";
+	else if (type == "js")
+		_response_map["Content-Type"] = "text/javascript";
+	else if (type == "jpeg" || type == "jpg")
+		_response_map["Content-Type"] = "image/jpeg";
+	else if (type == "png")
+		_response_map["Content-Type"] = "image/png";
+	else if (type == "bmp")
+		_response_map["Content-Type"] = "image/bmp";
+	else if (type == "ico")
+		_response_map["Content-Type"] = "image/x-icon";
+	else
+		_response_map["Content-Type"] = "text/plain";
+}
+
+void
+Response::construct_header_string(void)
+{
+	std::string CRLF = "\r\n";
+
+	_response_map["header-string"] += _response_map["Status-line"];
+	_response_map["header-string"] += CRLF;
+	_response_map["header-string"] += "Date: " + _response_map["Date"];
+	_response_map["header-string"] += CRLF;
+	_response_map["header-string"] += "Server: " + _response_map["Server"];
+	_response_map["header-string"] += CRLF;
+	_response_map["header-string"] += "Content-Length: " + _response_map["Content-Length"];
+	_response_map["header-string"] += CRLF;
+	_response_map["header-string"] += "Content-Type: " + _response_map["Content-Type"];
+	_response_map["header-string"] += CRLF;
+	_response_map["header-string"] += "Connection: " + _response_map["Connection"];
+	_response_map["header-string"] += CRLF;
+	_response_map["header-string"] += CRLF;
+}
+
+void
+Response::construct_body_string(std::string path_to_file)
+{
+	std::ifstream	  file;
+	std::stringstream buffer;
+
+	file.open(path_to_file.c_str());
+	if (file.fail())
+	{
+		std::cout << "Open file error" << std::endl;
+		return;
+	}
+	buffer << file.rdbuf();
+	_response_map["body-string"] = buffer.str();
+	file.close();
+}
+
+void
+Response::construct_full_response(void)
+{
+	_response_map["full-response-string"] += _response_map["header-string"];
+	_response_map["full-response-string"] += _response_map["body-string"];
+}
+
+std::string
+Response::get_http_response(void)
+{
+	std::string ret;
+	ret = _response_map["full-response-string"];
+	return ret;
+}
+
+void
+Response::create_error_html_page(int code)
+{
+	std::string html_page = "";
+	html_page += "<!DOCTYPE html><html><head><link rel=\"stylesheet\"href=\"style.css\"/><link rel=\"icon\" "
+				 "href=\"favicon.ico\"><title>";
+	html_page += std98::to_string(code);
+	html_page
+		+= "</title></head><body><div class=\" wrapper\"><div class=\"centered-box\"><h1 class=\"title\">";
+	html_page += _status_code.get_key_value_formated(code);
+	html_page += "</h1></div></div></body></html>";
+	_response_map["body-string"] = html_page;
+	std::cout << "AUTO_HTML PAGE:\n" << html_page << std::endl;
+}
+
+std::ostream &
+operator<<(std::ostream &output, Response const &res)
```

Comment:
Do we rename variable ?
```suggestion
operator<<(std::ostream &output, Response const &response)
```